extensions/msal-node-extensions/src/broker/NativeBrokerPlugin.ts

```diff
@@ -0,0 +1,415 @@
+    isBrokerAvailable: boolean;
+    
+    constructor() {
+        const defaultLoggerOptions: LoggerOptions = {
+            loggerCallback: () => {},
+            piiLoggingEnabled: false
+        };
+        this.logger = new Logger(defaultLoggerOptions, name, version); // Default logger
+        this.isBrokerAvailable = msalNodeRuntime.StartupError ? false : true;
+    }
+
+    setLogger(loggerOptions: LoggerOptions): void {
+        this.logger = new Logger(loggerOptions, name, version);
+        const logCallback = (message: string, logLevel: MsalRuntimeLogLevel, containsPii: boolean) => {
+            switch(logLevel) {
+                case MsalRuntimeLogLevel.Trace:
+                    if (containsPii) {
+                        this.logger.tracePii(message);
+                    } else {
+                        this.logger.trace(message);
+                    }
+                    break;
+                case MsalRuntimeLogLevel.Debug: 
+                    if (containsPii) {
+                        this.logger.tracePii(message);
+                    } else {
+                        this.logger.trace(message);
+                    }
+                    break;
+                case MsalRuntimeLogLevel.Info:
+                    if (containsPii) {
+                        this.logger.infoPii(message);
+                    } else {
+                        this.logger.info(message);
+                    }
+                    break;
+                case MsalRuntimeLogLevel.Warning:
+                    if (containsPii) {
+                        this.logger.warningPii(message);
+                    } else {
+                        this.logger.warning(message);
+                    }
+                    break;
+                case MsalRuntimeLogLevel.Error:
+                    if (containsPii) {
+                        this.logger.errorPii(message);
+                    } else {
+                        this.logger.error(message);
+                    }
+                    break;
+                case MsalRuntimeLogLevel.Fatal:
+                    if (containsPii) {
+                        this.logger.errorPii(message);
+                    } else {
+                        this.logger.error(message);
+                    }
+                    break;
+                default:
+                    if (containsPii) {
+                        this.logger.infoPii(message);
+                    } else {
+                        this.logger.info(message);
+                    }
+                    break; 
+            }
+        };
+        try {
+            msalNodeRuntime.RegisterLogger(logCallback, loggerOptions.piiLoggingEnabled);
+        } catch (e) {
+            const wrappedError = this.wrapError(e);
+            if (wrappedError) {
+                throw wrappedError;
+            }
+        }
+    }
+
+    async getAccountById(accountId: string, correlationId: string): Promise<AccountInfo> {
```

Comment:
I assume same here then? Public surface is still the same.

Reply:
This will be part of the public surface of `msal-node-extensions` but does not exist yet. This plugin will be consumed by `msal-node` and all of these APIs here will be called by the existing msal-node APIs. The public surface for `msal-node` will not be changing a whole lot besides a few new request/response parameters. PR with those changes is #5550